### PR TITLE
[rfc] [graphql-typescript-definitions] Update to handle multiple schemas

### PR DESCRIPTION
**Pull request by notjosh**
## Description

Fixes https://github.com/Shopify/quilt/issues/2367

This PR does two things:

- updates `graphql-typescript-definitions` to handle multiple schemas (both via array or glob), via `graphql-config` calls
- moves `graphql-typescript-definitions` toward not needing to depend on `graphql-config-utilities`

Most of the references to `graphql-config-utilities` have been removed in favour of calling equivalent methods on `graphql-config`. Most of them are one-liner replacements, with a couple of small helpers added.

The remaining references to `graphql-config-utilities` are for backward compatibility, as our usage of legacy configs isn't 100% backward compatible with modern `graphql-config` - i.e. `documents` is now used in favour of `includes` to refer to operations/fragments, but `project.getDocuments()` won't fall back to referring to `includes` in legacy projects, so we need to call `getGraphQLProjectIncludedFilePaths()` and `getIncludesExcludesFromConfig()`.

❓ We could fully remove `graphql-config-utilities` by inlining+modernising the two functions we use. None of our products depend on `graphql-config-utilities` directly, so it could be removed from Quilt. Thoughts?

❓ We could remove support for legacy configs entirely, and force legacy projects to update their config. `graphql-config@3` is 2 years old now, so I don't think it's completely unreasonable. However, I think plenty of teams still use the legacy format, so it would be a barrier to adopting new versions. Thoughts?

### CODE REVIEW

File: packages/graphql-typescript-definitions/loom.config.ts [hunk 1]
```diff
@@ -5,5 +5,9 @@ import {quiltPackage} from '../../config/loom';
 export default createPackage((pkg) => {
   pkg.entry({root: './src/index.ts'});
   pkg.binary({name: 'graphql-typescript-definitions', root: './src/cli.ts'});
-  pkg.use(quiltPackage());
+  pkg.use(
+    quiltPackage({
+      jestEnv: 'node',
```
Comment (notjosh) [inline packages/graphql-typescript-definitions/loom.config.ts:10]: Works around this error: https://github.com/prisma/prisma/issues/8558

File: packages/graphql-typescript-definitions/src/index.ts [hunk 2]
```diff
@@ -407,11 +399,16 @@ export class Builder extends EventEmitter {
...
-      this.emit('error', error);
+    } catch (error) {
+      if (error instanceof Error) {
+        const wrapped = new Error(
+          `Error in ${
+            file.path
+          }: ${error.message[0].toLowerCase()}${error.message.slice(1)}`,
+        );
+        this.emit('error', wrapped);
+      }
```
Comment (notjosh) [inline packages/graphql-typescript-definitions/src/index.ts:410]: TypeScript was angry, so this became a little more generic.

**Review by BPScott — COMMENTED** [review]
Nice start! This suggests that migrating this to handle `documents` and multiple schemas in addition to `include`/`exclude` keys wouldn't be toooo complex.

I think there's some confusion over the notion of "legacy". graphql-config papers over the notion of legacy names (includes / excludes plural, replacing them with their singular versions), we don't need to do anything to handle them so there's nothing to drop support for.


We need to maintain support for using the `include` and `exclude` keys - as they are not legacy. We need to combine the "documents" and "include/exclude" arrays not pick one or the other. I've detailed more about this below.

I think this got confusing because our fixtures use the legacy names  (includes / excludes plural, schemaPath), but we lack any fixtures that use the modern names (include / exclude singular, schema), and thus you you conflated the idea of "include/exclude" as being a legacy concept when it is not.

**Comment by BPScott** [comment]
> ❓ We could fully remove `graphql-config-utilities` by inlining+modernising the two functions we use. None of our products depend on `graphql-config-utilities` directly, so it could be removed from Quilt. Thoughts?

I like this long-term direction. In order to fully deprecate and remove graphql-config-utilities we'd need to also remove it from `graphql-fixtures` in quilt, and sewing-kit according to https://grokt.shopify.io/results?q=graphql-config-utilities+-r%3AShopify%2Fquilt
It feels like we're getting close to doing too many things at once - are we trying to re


> ❓ We could remove support for legacy configs entirely, and force legacy projects to update their config. `graphql-config@3` is 2 years old now, so I don't think it's completely unreasonable. However, I think plenty of teams still use the legacy format, so it would be a barrier to adopting new versions. Thoughts?

Per stuff above, we don't deal with the normalisation from legacy to modern config styles - graphql-config handles that. We still need to maintain support for the modern `include` / `exclude` keys

**Pull request closed without merging**